docs/design/configuration.md

```diff
@@ -0,0 +1,141 @@
+# Kubernetes Component Configuration
+
+## Abstract
+
+A generic ``Configuration`` object to store command-line flags that is used for
```

Comment:
Especially because all of those could be represented as files (sysconfig or
env file).

On Jul 23, 2015, at 7:40 PM, Brian Grant <notifications@github.com> wrote:

In docs/design/configuration.md
<https://github.com/GoogleCloudPlatform/kubernetes/pull/10982#discussion_r35384137>
:

> @@ -0,0 +1,141 @@
> +# Kubernetes Component Configuration
> +
> +## Abstract
> +
> +A generic ``Configuration`` object to store command-line flags that is used for

That said, I don't think this should be only for Kubernetes configuration.

Applications commonly consume configuration via:

   - command-line flags
   - environment variables
   - configuration files

All should be possible, but we should focus on files first.

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/10982/files#r35384137>
.